test/integration/cluster_filter_integration_test.cc

```diff
@@ -10,32 +11,48 @@
 namespace Envoy {
 namespace {
 
+// Helper class for passing runtime information from a test filter to the test class.
+// May only be called from one filter during a test.
+class TestParent {
+public:
+  virtual ~TestParent() = default;
+
+  // Called from a worker thread
+  virtual void onNewConnectionCalled(bool on_write_called) PURE;
+};
+
 class PoliteFilter : public Network::Filter, Logger::Loggable<Logger::Id::filter> {
 public:
-  PoliteFilter(const ProtobufWkt::StringValue& value) : greeting_(value.value()) {}
+  PoliteFilter(TestParent& parent, const ProtobufWkt::StringValue& value)
+      : parent_(parent), greeting_(value.value()) {}
 
   Network::FilterStatus onData(Buffer::Instance& data, bool end_stream) override {
-    ENVOY_CONN_LOG(debug, "polite: onData {} bytes {} end_stream", read_callbacks_->connection(),
+    ENVOY_CONN_LOG(warn, "polite: onData {} bytes {} end_stream", read_callbacks_->connection(),
                    data.length(), end_stream);
     if (!read_greeted_) {
-      Buffer::OwnedImpl greeter(greeting_);
-      read_callbacks_->injectReadDataToFilterChain(greeter, false);
+      if (greeting_.length() > 0) {
+        Buffer::OwnedImpl greeter(greeting_);
+        read_callbacks_->injectReadDataToFilterChain(greeter, false);
+      }
       read_greeted_ = true;
     }
     return Network::FilterStatus::Continue;
   }
   Network::FilterStatus onWrite(Buffer::Instance& data, bool end_stream) override {
-    ENVOY_CONN_LOG(debug, "polite: onWrite {} bytes {} end_stream", write_callbacks_->connection(),
+    ENVOY_CONN_LOG(warn, "polite: onWrite {} bytes {} end_stream", write_callbacks_->connection(),
```

Comment:
can we revert log changes?

Reply:
Yes, they are all in the last commit I added for debugging.